sys-tuner/src/main.rs

```diff
@@ -0,0 +1,62 @@
+use std::fs;
+use unix_socket::UnixListener;
+
+#[cfg(any(not(unix), target_os = "macos"))]
+pub const SOLANA_SYS_TUNER_PATH: &str = "/tmp/solana-sys-tuner";
+
+// Use abstract sockets for Unix builds
+#[cfg(all(unix, target_os = "linux"))]
+pub const SOLANA_SYS_TUNER_PATH: &str = "\0/tmp/solana-sys-tuner";
+
+#[allow(dead_code)]
+#[cfg(target_os = "linux")]
+fn tune_system() {
+    use std::process::Command;
+    use std::str::from_utf8;
+    use thread_priority::*;
+
+    let output = Command::new("ps")
+        .arg("-eT")
+        .output()
+        .expect("Expected to see all threads");
+
+    if output.status.success() {
+        if let Ok(threads) = from_utf8(&output.stdout) {
+            let mut threads: Vec<&str> = threads.split('\n').collect();
+            for t in threads.iter_mut() {
+                if t.find("solana-poh-ser").is_some() {
+                    let pids: Vec<&str> = t.split_whitespace().collect();
+                    let thread_id = pids[1].parse::<u64>().unwrap();
+                    let _ignored = set_thread_priority(
+                        thread_id,
+                        ThreadPriority::Max,
+                        ThreadSchedulePolicy::Realtime(RealtimeThreadSchedulePolicy::Fifo),
+                    );
+                    break;
+                }
+            }
+        }
+    }
+}
+
+#[allow(dead_code)]
+#[cfg(any(not(unix), target_os = "macos"))]
+fn tune_system() {}
+
+#[allow(dead_code)]
+fn main() {
+    solana_logger::setup();
```

Comment:
We should restrict who can use this daemon better.  For example right now I could get real time priority for my program, even if I'm not a `solana-validator` process by naming a thread "solana-poh-serv".  

There could be a command-line argument to sys-tuner that indicates the expected uid of the `solana-validator` process and ignore all "solana-poh-serv" threads that don't belong to that user, to at least slightly lock this down

Reply:
Abstract namespace sockets don't manifest in the filesystem. So, we cannot set file permissions for it. Actually, I had implemented the code to tweak the file permissions, but it failed on Linux for above reason. I can use regular (non-Abstract) sockets. That'll allow us to set correct group permissions.

Reply:
Oh!  How does one set permission on an abstract namespace socket?

Reply:
One way is to check the peers UID/GID after they connect to the socket. But, that's post connection.

Reply:
That would be fine

Reply:
For now, I have switched to using named socket. That also cleaned up the remove old socket workflow.